### PR TITLE
tests: enhance xtimer_msg_receive_timeout

### PR DESCRIPTION
**Pull request by smlng**
- correct test period from 100ms to 1000ms, to match description
    - moved description to separate README.md
    - changed timer message type from 44 to 42, for nerdiness
    - minor code enhancements

### CODE REVIEW

File: tests/xtimer_msg_receive_timeout/main.c [hunk 1]
```diff
@@ -30,27 +26,27 @@
 #include "xtimer.h"
 #include "timex.h"
 
-#define TEST_PERIOD (100000LU)
+#define TEST_PERIOD     (1*US_PER_SEC) /* 1 sec in US */
```
Comment (kaspar030) [inline tests/xtimer_msg_receive_timeout/main.c]: 1U *edit* sorry, ```1LU``` please.

File: tests/xtimer_msg_receive_timeout/main.c [hunk 2]
```diff
@@ -30,27 +26,27 @@
...
+#define TEST_COUNT      (10)
```
Comment (kaspar030) [inline tests/xtimer_msg_receive_timeout/main.c]: ```10LU``` please. C uses the left hand type to determine result type, and a number literals default to ```int```. So if later down you're using "10 + offset", the result will overflow at 2^16 on 16bit platforms. *edit* Actually it should overflow at ```2^15```.

File: tests/xtimer_msg_receive_timeout/main.c (hunk 1 above)
Comment (kaspar030) [inline tests/xtimer_msg_receive_timeout/main.c]: Btw, does the test still work if you lower the time to 100ms? Having automatic testing in mind, the shorter, the better, so the shortest to actually show any bugs would be perfect.

Comment (smlng) [inline tests/xtimer_msg_receive_timeout/main.c]: mhm, will test - however, the I better make this `1000LU * US_PER_MS` so it can be adapted more easily?

Comment (smlng) [inline tests/xtimer_msg_receive_timeout/main.c]: seems to work, so default to faster (100ms) or slower (1000ms) variant?

Comment (kaspar030) [inline tests/xtimer_msg_receive_timeout/main.c]: Could you test on a device with a slower UART? e.g., arduino-mega2580? (Does the time needed for output even change the actual test timing?)

Comment (smlng) [inline tests/xtimer_msg_receive_timeout/main.c]: hey, only have an _duemilanove_ at hand. However, that works with 100ms, too. Drift in output timing is around 10ms late from the expected timestamp, but works.

**Comment by smlng** [comment]
@kaspar030 addressed your comments, happy? So I can squash and have murdock take a look ...

**Comment by smlng** [comment]
@kaspar030 I squashed and give murdock a try ... you happy too and care to Ack'n'Merge?

**Comment by smlng** [comment]
ping @kaspar030

**Comment by smlng** [comment]
ping again

File: tests/xtimer_msg_receive_timeout/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# test application for xtimer_msg_receive_timeout()
+
+This test will start sequentially start TEST_COUNT xtimers to send a IPC msg,
```
Comment (cgundogan) [inline tests/xtimer_msg_receive_timeout/README.md]: one `start` too much

File: tests/xtimer_msg_receive_timeout/main.c [hunk 4]
```diff
@@ -30,27 +26,27 @@
...
         if (xtimer_msg_receive_timeout(&m, TEST_PERIOD) < 0) {
             puts("Timeout!");
         }
         else {
-            printf("Message received: %" PRIu16 "\n", m.type);
+            printf("Message: %" PRIu16 "\n", m.type);
         }
-        offset = (offset < 0) ? 1000 : -1000;
-        xtimer_remove(&t);
+        /* flip signedness */
```
Comment (kaspar030) [inline tests/xtimer_msg_receive_timeout/main.c]: minor one: "flip sign". "signedness" is a property of a type, not of a value

**Comment by cgundogan** [comment]
Murdock agrees - GO

**Pull request merged**